tests/components/advantage_air/test_binary_sensor.py

```diff
@@ -0,0 +1,69 @@
+"""Test the Advantage Air Binary Sensor Platform."""
+
+from homeassistant.const import STATE_OFF, STATE_ON
+
+from tests.components.advantage_air import (
+    TEST_SET_RESPONSE,
+    TEST_SET_URL,
+    TEST_SYSTEM_DATA,
+    TEST_SYSTEM_URL,
+    add_mock_config,
+)
+
+
+async def test_binary_sensor_async_setup_entry(hass, aioclient_mock):
+    """Test binary sensor setup."""
+
+    aioclient_mock.get(
+        TEST_SYSTEM_URL,
+        text=TEST_SYSTEM_DATA,
+    )
+    aioclient_mock.get(
+        TEST_SET_URL,
+        text=TEST_SET_RESPONSE,
+    )
+    await add_mock_config(hass)
+
+    registry = await hass.helpers.entity_registry.async_get_registry()
+
+    assert len(aioclient_mock.mock_calls) == 1
+
+    # Test First Air Filter
+    entity_id = "binary_sensor.ac_one_filter"
+    state = hass.states.get(entity_id)
+    assert state
+    assert state.state == STATE_OFF
+
+    entry = registry.async_get(entity_id)
+    assert entry
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)